blocks/editable/index.js

```diff
@@ -68,18 +68,26 @@ export default class Editable extends Component {
 		this.onSelectionChange = this.onSelectionChange.bind( this );
 		this.maybePropagateUndo = this.maybePropagateUndo.bind( this );
 		this.onPastePostProcess = this.onPastePostProcess.bind( this );
+		this.onToolbarAddLink = this.onToolbarAddLink.bind( this );
+		this.onToolbarEditLink = this.onToolbarEditLink.bind( this );
+		this.onToolbarChangeLinkValue = this.onToolbarChangeLinkValue.bind( this );
 
 		this.state = {
 			formats: {},
-			bookmark: null,
 			empty: ! props.value || ! props.value.length,
+			toolbar: {
+				isAddingLink: false,
+				isEditingLink: false,
+				newLinkValue: '',
+			},
```

Comment:
Why can't we keep this state in the `formatToolbar` instead. Seems more related to the link modal than the Editable.